Tasks/ServiceFabricComposeDeploy/ServiceFabricComposeDeploy.ps1

```diff
@@ -0,0 +1,116 @@
+        'Compose' = $composeFilePath
+    }
+    $removeParameters = @{
+        'Force' = $true
+    }
+    $getStatusParameters = @{
+        'ApplicationName' = $applicationName
+    }
+
+    # Connect to the cluster
+    $clusterConnectionParameters = @{}
+    Connect-ServiceFabricClusterFromServiceEndpoint -ClusterConnectionParameters $clusterConnectionParameters -ConnectedServiceEndpoint $connectedServiceEndpoint
+
+    $repositoryCredentials = Get-VstsInput -Name repositoryCredentials -Require
+    switch ($repositoryCredentials) {
+        "Endpoint"
+        {
+            $dockerRegistryEndpointName = Get-VstsInput -Name dockerRegistryEndpointName -Require
+            $dockerRegistryEndpoint = Get-VstsEndpoint -Name $dockerRegistryEndpointName -Require
+            $authParams = $dockerRegistryEndpoint.Auth.Parameters
+            $username = $authParams.username
+            $password = $authParams.password
+            $isEncrypted = $false
+        }
+        "UsernamePassword"
+        {
+            $username = Get-VstsInput -Name repositoryUserName -Require
+            $password = Get-VstsInput -Name repositoryPassword -Require
+            $isEncrypted = (Get-VstsInput -Name passwordEncrypted -Require) -eq "true"
+        }
+    }
+
+    if ($repositoryCredentials -ne "None")
+    {
+        if ((-not $isEncrypted) -and $connectedServiceEndpoint.Auth.Parameters.ServerCertThumbprint)
+        {
+            $thumbprint = $connectedServiceEndpoint.Auth.Parameters.ServerCertThumbprint
+
+            $cert = Get-Item -Path "Cert:\CurrentUser\My\$thumbprint" -ErrorAction SilentlyContinue
+            if($cert -ne $null)
+            {
+                Write-Host (Get-VstsLocString -Key EncryptingPassword)
+                $password = Invoke-ServiceFabricEncryptText -Text $password -CertStore -CertThumbprint $thumbprint -StoreName "My" -StoreLocation CurrentUser
+                $isEncrypted = $true
+            }
+            else
+            {
+                Write-Host (Get-VstsLocString -Key CertificateNotFound)
+            }
+        }
+
+        $deployParameters['RepositoryUserName'] = $username
+        $deployParameters['RepositoryPassword'] = $password
+        $deployParameters['PasswordEncrypted'] = $isEncrypted
+    }
+
+    if ($deployTimeoutSec)
+    {
+        $deployParameters['TimeoutSec'] = $deployTimeoutSec
+    }
+    if ($removeTimeoutSec)
+    {
+        $removeParameters['TimeoutSec'] = $removeTimeoutSec
+    }
+    if ($getStatusTimeoutSec)
+    {
+        $getStatusParameters['TimeoutSec'] = $getStatusTimeoutSec
+    }
+
+    $existingApplication = Get-ServiceFabricDockerComposeApplicationStatusPaged @getStatusParameters
+    if ($existingApplication -ne $null)
+    {
+        Write-Host (Get-VstsLocString -Key RemovingApplication -ArgumentList $applicationName)
+        $removeParameters['ApplicationName'] = $applicationName
+        Remove-ServiceFabricDockerComposeApplication @removeParameters
```

Comment:
Yes it does mean that, and is a large part of why this is only a Preview. It is not production ready and that is something we will need to make clear in the documentation.